subsystem/kubeapi_test.go

```diff
@@ -424,14 +424,21 @@ func getDefaultNMStateConfigSpec(nicPrimary, nicSecondary, macPrimary, macSecond
 	}
 }
 
-func getAgentMac(agent *v1beta1.Agent) string {
-
-	for _, agentInterface := range agent.Status.Inventory.Interfaces {
-		if agentInterface.MacAddress != "" {
-			return agentInterface.MacAddress
+func getAgentMac(ctx context.Context, client k8sclient.Client, key types.NamespacedName) string {
+	mac := ""
+	Eventually(func() bool {
+		agent := getAgentCRD(ctx, client, key)
+		for _, agentInterface := range agent.Status.Inventory.Interfaces {
+			if agentInterface.MacAddress != "" {
+				fmt.Println("Found mac !!!!!")
```

Comment:
maybe replace with a log
and with less panic :)

Reply:
oops